### ISSUE TITLE
Input layers support (Button, TextInput, etc...)

### ISSUE THREAD

**Issue by GuillaumeSalles** [airbnb/Lona#45]
I don't think it was already asked or mentioned in the README, but do you plan to support inputs like buttons, checkboxes, text input, etc? If yes, how do you see it? 

I didn't really thought about the implementation but I imagine it will be challenging to generate components inputs that share the same behaviors/styles on different platforms.

**Comment by dabbott** [comment]
I definitely want to support these. I agree it'll probably be tricky... I haven't attempted to this, since we haven't needed them for internal projects just yet (but we will). Here's what I'm thinking so far. 

Let's take a Text Input for example:

1. There will be a "lowest common denominator" kind of Text Input, that supports a limited variety of parameters, but looks (mostly) the same across all platforms. There may be some platform-specific parameters (e.g. underline on Android). This is essentially the React Native approach. In Lona Studio, you'll be able to preview and interact with this component.

2. This component (and _all components_) can be configured to use an _existing implementation_ instead of the implementation in the `.component` file. So instead of Lona generating code or custom Views/divs, it will instantiate an existing component/class. Parameters will still be defined/passed the same as other component. In Lona Studio, you won't see an accurate preview of the component. You'll see the underlying `.component` -- you can use the "lowest common denominator" or even a totally different component that's just an image. It's akin to the Xcode way of specifying the custom class of a View:

![screen shot 2017-12-14 at 8 42 35 am](https://user-images.githubusercontent.com/1198882/34003685-c7712c76-e0aa-11e7-95ef-3b87c558211e.png)

The 2nd case is important in general, since often we want to delegate to existing implementations on the device (since they've been hand-tuned for perf, or have very complex or platform-specific behavior).

That's my thinking so far. What are your thoughts?

**Comment by GuillaumeSalles** [comment]
I fully agree with your proposition. A little bit like the React Native mantra ("learn once, write anywhere"), learn Lona once, build components anywhere. 

Lona already supports properties that are not natively supported in web development (Aspect ratio, max number of lines). I can imagine a `Targeted Platforms` option on the component metadata if the lowest common denominator is not enough.

**Comment by vasicvuk** [comment]
Is there a plan for this in recent future?
